nixos/modules/services/networking/nanod.nix

```diff
@@ -0,0 +1,106 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.nanod;
+
+in {
+
+  options.services.nanod = {
+
+    enable = mkEnableOption "nanod";
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/nano";
+      description =
+        "The data directory where nanod will store its state, including the block-lattice";
+    };
+
+    nodeConfig = mkOption {
+      type = types.path;
+      description =
+        "The path to the config-node.toml file. Setting this option will overwrite any other configuration setting";
+      default = pkgs.runCommand "config-node.toml" {
+        json = builtins.toJSON cfg.nodeConfigOptions;
+        passAsFile = [ "json" ];
+      } ''
+        ${pkgs.remarshal}/bin/json2toml < $jsonPath > $out
+      '';
+      defaultText =
+        "Generates a TOML configuration from services.nanod.nodeConfigOptions";
+    };
+
+    nodeConfigOptions = mkOption {
+      type = types.attrs;
+      description =
+        "Configuration options for the nano node daemon. See: https://docs.nano.org/running-a-node/configuration/";
+      example = {
+        node = { enable-voting = true; };
+        websocket = {
+          address = "::1";
+          enable = true;
+          port = 7078;
+        };
+      };
+    };
+
+    rpcConfig = mkOption {
```

Comment:
Might be more clear to name as: `rpcConfig` -> `rpcConfigFile` and `rpcConfigOptions` -> `rpcConfig`... but just my opinion, take it or leave it :man_shrugging: